pkg/network/tracer/offsetguess/tracer.go

```diff
@@ -698,6 +703,22 @@ func (t *tracerOffsetGuesser) flowi6EntryState() GuessWhat {
 	return GuessSAddrFl6
 }
 
+func setupLoopbackInNS(ns netns.NsHandle) error {
+	if err := netns.Set(ns); err != nil {
```

Comment:
Can we just wrap the call to `setupLookpbackInNS` in `WithNS` instead of explicitly locking here (or just wrap the contents here in `WithNS`)? `WithNS` will only lock if `ns` is not the current namespace.

Reply:
so I actually don't think it's necessary to set the NS here at all. Before calling this method we call `netns.New()`, which creates a new NS `and` sets it as active. I believe that also answers your question below about why we are calling `unix.Setns(int(curNS), unix.CLONE_NEWNET)` (to go back)

Reply:
Well that is kind of sneaky. I suspect the `clone` syscall doesn't really give you any other option but to switch to the new ns. Can you put some comments in the code clarifying what is going on here (and further down below)?

Reply:
yeah (and I think you mean the `.new` syscall) it doesn't have any option to not switch sadly.